From ff2d9f8143326d22211aae5e4b69333c079ddae3 Mon Sep 17 00:00:00 2001
From: Tomas Krupka <6817216+krupkat@users.noreply.github.com>
Date: Thu, 8 Jun 2023 17:42:13 +0200
Subject: [PATCH] fix build on arm

---
 xpano/algorithm/multiblend.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/xpano/algorithm/multiblend.cc b/xpano/algorithm/multiblend.cc
index 7ffd7c7..0c0cae7 100644
--- a/xpano/algorithm/multiblend.cc
+++ b/xpano/algorithm/multiblend.cc
@@ -34,7 +34,8 @@ void SafeMemset(uint8_t *ptr, uint8_t value, size_t num, const uint8_t *end) {
 
 // Convert from Multiblend's Flex format to OpenCV's UMat.
 // Flex is a RLE format, leftmost bit is the mask flag, the rest is the length.
-cv::UMat ToUMat(multiblend::utils::Flex &flex) {
+template <typename TFlexType>
+cv::UMat ToUMat(TFlexType &flex) {
   auto mask = cv::Mat(flex.height_, flex.width_, CV_8U);
 
   flex.Start();
-- 
2.34.1

